core/test/PrognosticDataIO_test.cpp

```diff
@@ -0,0 +1,154 @@
+/*!
+ * @file PrognosticData_test.cpp
+ *
+ * @date 7 Sep 2023
+ * @author Tim Spain <timothy.spain@nersc.no>
+ */
+
+#include "include/PrognosticData.hpp"
+
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include <doctest/doctest.h>
+
+#include "include/PrognosticData.hpp"
+
+#include "include/ConfiguredModule.hpp"
+#include "include/IStructure.hpp"
+#include "include/ModelComponent.hpp"
+#include "include/Module.hpp"
+#include "include/StructureFactory.hpp"
+#include "include/UnescoFreezing.hpp"
+#include "include/constants.hpp"
+
+#include <filesystem>
+#include <sstream>
+
+extern template class Module::Module<Nextsim::IOceanBoundary>;
+
+namespace Nextsim {
+
+// The CMake files uses TRUE and FALSE. Reduce these (and the CMake defaults ON & OFF) to C++ standard values
+#ifndef TRUE
+#define TRUE true
+#endif
+#ifndef FALSE
+#define FALSE false
+#endif
+#define ON true
+#define OFF false
+
+TEST_SUITE_BEGIN("PrognosticDataIO");
+// Once MPI-enabled IO is fully working, integrate this test into PrognosticData_test.cpp
```

Comment:
Flag this with `TODO` prefix so it's more easily `grep`-able.

Reply:
Added the TODO marker.